protocol/river-config-unstable-v1.xml

```diff
@@ -0,0 +1,115 @@
+    WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+    MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+  </copyright>
+
+  <interface name="zriver_config_v1" version="1">
+    <description summary="store and retrieve options">
+    </description>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the zriver_config_v1 object">
+        This request indicates that the client will not use the
+        zriver_config_v1 object any more. Objects that have been created
+        through this instance are not affected.
+      </description>
+    </request>
+
+    <enum name="value_type">
+      <entry name="int" value="0"/>
+      <entry name="uint" value="1"/>
+      <entry name="fixed" value="2"/>
+      <entry name="string" value="3"/>
+    </enum>
+
+    <request name="declare_option">
+      <description summary="declare a new option">
+        Declare a new option with a given type. This may silently fail if
+        the option already exists. The initial value of the option depends
+        on the type: for numeric types the initial value is 0, for strings
+        the initial value is NULL.
+      </description>
+      <arg name="key" type="string"/>
+      <arg name="type" type="uint" enum="value_type"/>
+    </request>
+
+    <request name="get_option">
+      <arg name="key" type="string"/>
+      <arg name="option" type="new_id" interface="zriver_option_v1"/>
+    </request>
+  </interface>
+
+  <interface name="zriver_option_v1" version="1">
+    <description summary="">
+      On binding this object, one of the event will immediately be sent by
+      the server. If the event is no_such_option, the client may do nothing
+      but destroy the object with the destroy request. Otherwise the type
+      of value sent indicates the immutable type of the option. Making a
+      set value request of a different type is a protocol error.
+    </description>
+
+    <enum name="error">
+      <entry name="type_mismatch" value="0"
+        summary="the client made a set value request of the wrong type."/>
+    </enum>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the zriver_option_v1 object">
+        This request indicates that the client will not use the
+        zriver_option_v1 object any more.
+      </description>
+    </request>
+
+    <event name="no_such_option">
+      <description summary="">
+        After this is emitted, no further events will be sent. Upon receiving
+        this, the client should make the destroy request.
+      </description>
+    </event>
```

Comment:
We could actually get rid of this request and make this a protocol error as the client may always call `declare_option` first.